Framework/CurveFitting/src/Algorithms/SplineInterpolation.cpp

```diff
@@ -192,28 +188,38 @@ void SplineInterpolation::exec() {
     }
 
     for (size_t i = 0; i < histNo; ++i) {
-      // set up the function that needs to be interpolated
-      std::unique_ptr<gsl_interp_accel, void (*)(gsl_interp_accel *)> acc(gsl_interp_accel_alloc(),
-                                                                          gsl_interp_accel_free);
-      std::unique_ptr<gsl_interp, void (*)(gsl_interp *)> linear(gsl_interp_alloc(gsl_interp_linear, binsNo),
-                                                                 gsl_interp_free);
-      gsl_interp_linear->init(linear.get(), &(iwspt->x(i)[0]), &(iwspt->y(i)[0]), binsNo);
-
       // figure out the interpolation range
       const std::pair<size_t, size_t> range = findInterpolationRange(iwspt, mwspt, i);
 
+      // set up the function that needs to be interpolated
+      Mantid::Kernel::LinearSpline<double, double> spline(iwspt->x(i).rawData(), iwspt->y(i).rawData());
+
       // perform interpolation in the range
-      for (size_t k = range.first; k < range.second; ++k) {
-        gsl_interp_linear->eval(linear.get(), &(iwspt->x(i)[0]), &(iwspt->y(i)[0]), binsNo, mwspt->x(0)[k], acc.get(),
-                                &(outputWorkspace->mutableY(i)[k]));
-        // calculate only 1st order derivative if needed
-        if (order > 0) {
-          gsl_interp_linear->eval_deriv(linear.get(), &(iwspt->x(i)[0]), &(iwspt->y(i)[0]), binsNo, mwspt->x(0)[k],
-                                        acc.get(), &(derivs[i]->mutableY(0)[k]));
-        }
-      }
+      // NOTE for legacy compatibility, this uses only the FIRST spectrum's X-axis for all other interpolations
+      std::vector<double> yNew(mwspt->x(0).size());
+      std::span<double const> xInRange(mwspt->x(0).cbegin() + range.first, mwspt->x(0).cbegin() + range.second);
```

Comment:
So you know, you can do this as (begin_iter, size)
```suggestion
      std::span<double const> xInRange(mwspt->x(0).cbegin() + range.first, range.second - range.first);
```